pykit/tests/test_command.py

```diff
@@ -0,0 +1,38 @@
+import unittest
+import os
+from pykit.transport.http import connect
+from pykit import Command
+
+
+class TestCommand(unittest.TestCase):
+    def setUp(self):
+        self.connection = connect(
+            os.environ['PK_DB2SOCK_URL'],
+            db2sock_auth=(
+                os.environ['PK_DB2SOCK_USER'], os.environ['PK_DB2SOCK_PASS']))
+
+    def test_execute_qsh_command(self):
+        cmd = Command("ls -1 /QOpenSys", True)
```

Comment:
Command calls a QSH command? I thought this was for CL commands?

Reply:
It can call QSH, REXX, or EXEC depending on the options.

I thought we talked about this?

Reply:
When calling the Command class, I'd except it to only call a CL Command. It might use the `qsh`-style call under the covers, though. I'd expect something like in the old toolkit, where it wrappers the command in a call to `system`: https://bitbucket.org/litmis/python-itoolkit/src/be890375edb4a9d98a3beca7e96956604720e2da/itoolkit/itoolkit.py#lines-342

Reply:
Jesse requested these three. I'd like to see what he thinks.

Reply:
No reason to call REXX from Python, in my opinion. No one has ever asked to call REXX.

Reply:
It was used in the PHP toolkit and XMLSERVICE.

Reply:
Although REXX is slow, so it might have been better to force people to compile RTV* commands into ordinary CL programs and call those CL programs to avoid the performance issues.

Reply:
Tony's "REXX" nomenclature may have been misleading. We pass in an RTV* command and specify "REXX," and then XMLSERVICE used REXX under the covers to run the command. It's not like we provide REXX code to run. REXX was an implementation detail of XMLSERVICE that was exposed in the keyword 'rexx' :(

Reply:
Maybe that's obvious. Just stating for the record.

Reply:
PHP toolkit used several specific methods to do regular commands and QSH, REXX, so the user had to choose the method and know what he/she was doing. Here, Josh appears to be using some "AI" to guess the user's intent. The old way required knowledge and explanation. I suppose the new way could be fine-tuned over time, such as looking for a whitelist of RTV* commands? Later...?

Reply:
Thanks for the input @alanseiden. That's exactly what we had discussed early on. Kevin, Jesse, and yourself pointed me to resources about the differences between REXX and EXEC. I had read up, and I *thought* we had decided that Command would handle QSH, REXX, and EXEC, depending on the input for Command.

Reply:
I told Tony once that it wouldn't be too difficult to bypass REXX and call the CL command directly. You just need to call an API to get the parameter descriptions for the CL command processing program. Then you just set up your parameter storage and call the command processing program like you would any other program.

His response: "You shouldn't be doing output parameters anyway, so I don't care that it's slow"

Reply:
@alanseiden I agree that the naming is misleading. "rexx" is used to get output parameters while "exec" is a standard CL command call. Lastly there is "qsh" exec mode, which really is used to call a CL command through the `system` command in `QSH`, which allows one to get the 5250 output equivalent returned. While this goes through QSH, the intent is not to call a QSH command (though one is used internally), it's to call a CL command.

So I'm not disagreeing that all 3 exec modes should be handled, it's just that Tony liked to confuse people about their intent by exposing cryptic implementation details to the user:

- "exec": standard CL command call through `QCMDEXC` (no return data)
- "rexx": call CL command with output parameters (return parameters returned)
- "qsh": call CL command using QSH `system` command (return screen output)

Reply:
Yes, Tony objected to the idea of the RTV* commands and their output parameters. "They designed it wrong."